test/functional/test_framework/test_framework.py

```diff
@@ -203,67 +197,60 @@ def main(self):
 
     # Public helper methods. These can be accessed by the subclass test scripts.
 
-    def start_node(self, i, dirname, extra_args=None, rpchost=None, timewait=None, binary=None, stderr=None):
+    def start_node(self, i, dirname, extra_args=[], rpchost=None, timewait=None, binary=None, stderr=None):
```

Comment:
`null` in Python is `None`, so yeah, I think `None` should be the default here and just set it to `[]` if it's `None` at the start.